docs: add WASM config for Apache and Nginx

RanadeepPolavarapu opened the pull request:
Resolves #1544.

Signed-off-by: Ranadeep Polavarapu <ranadeep@icloud.com>

GitHub Issue (If applicable): #1544.

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal) -->

## PR Type

What kind of change does this PR introduce?
<!-- Please uncomment one or more that apply to this PR

- Bugfix
- Feature
- Code style update (formatting)
- Refactoring (no functional changes, no api changes)
- Build or CI related changes
- Documentation content changes
- Project automation
- Other... Please describe:

-->
- Documentation content changes (New documentation)

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Adds documentation on how to configure a web server for reverse proxying a WASM application, specifically for Uno.

## What is the new behavior?

<!-- Please describe the new behavior after your modifications. -->


## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

## Other information

<!-- Please provide any additional information if necessary -->

Internal Issue (If applicable):
<!-- Link to relevant internal issue if applicable. All PRs should be associated with an issue (GitHub issue or internal) -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/unoplatform/uno?pullRequest=4246) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/unoplatform/uno?pullRequest=4246) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/unoplatform/uno?pullRequest=4246) it.</sub>

RanadeepPolavarapu commented:
Just adding this snippet below to make it clear and easier for the reviewer when reviewing the *nginx* portion of the document.

![image](https://user-images.githubusercontent.com/7084995/95665074-f7d63d00-0b1a-11eb-8d21-b990c819a1ed.png)


https://github.com/unoplatform/uno/pull/4246/files#diff-131cc6567446afc10e20bc888c7f317bR58-R64.

jeromelaban commented on doc/articles/getting-started/deployments.md:
```diff
@@ -0,0 +1,167 @@
+
+-   `application/wasm`
+-   `application/octet-stream`
+-   `application/octet-stream`
```
Duplicate ?

RanadeepPolavarapu replied:
Yes, I have removed the duplicate.

jeromelaban commented on doc/articles/getting-started/deployments.md:
```diff
@@ -0,0 +1,167 @@
+-   `application/octet-stream`
+-   `application/octet-stream`
+-   `application/font-woff`
+-   `application/font-woff`
```
Duplicate ?

jeromelaban commented on doc/articles/getting-started/deployments.md:
```diff
@@ -0,0 +1,167 @@
+  # faster then read() + write()
+  sendfile on;
+
+  # send headers in one peace, its better then sending them one by one
```
```suggestion
  # Send headers in one piece, it's better then sending them one by one
```

jeromelaban commented on doc/articles/getting-started/deployments.md:
```diff
@@ -0,0 +1,167 @@
+  # don't buffer data sent, good for small data bursts in real time
+  tcp_nodelay on;
+
+  # reduce the data that needs to be sent over network -- for testing environment
```
Why is this test environment only ? Wasm apps benefit a lot from compression, and even better if brotli can be enabed.

RanadeepPolavarapu replied:
Added Brotli & removed test environment comment.

jeromelaban commented on doc/articles/getting-started/deployments.md:
```diff
@@ -0,0 +1,167 @@
+  # Security reasons, turn off nginx versions
+  server_tokens off;
+
+  more_clear_headers Server; # Custom module: headers-more-nginx-module (https://github.com/openresty/headers-more-nginx-module)
```
Is this a default configuration or do users need to install something additional ?

RanadeepPolavarapu replied:
Removed this as it requires an additional module.

jeromelaban commented on doc/articles/getting-started/deployments.md:
```diff
@@ -0,0 +1,167 @@
+# WASM Web Server Configuration
```
```suggestion
# How to host a WebAssembly App
```

jeromelaban commented on doc/articles/getting-started/deployments.md:
```diff
@@ -0,0 +1,167 @@
+  - [Apache](#apache)
+
+
+Regardless of what webserver (reverse proxy) you are using, you will at core need to support the following Content (MIME) types.
```
```suggestion
Regardless of the webserver (or reverse proxy) software used, the support the following Content (MIME) types are always needed:
```

jeromelaban commented on doc/articles/getting-started/deployments.md:
```diff
@@ -0,0 +1,167 @@
+
+## Nginx
+
+Below contains a extremly performance tuned version of *nginx* fitting WASM deployments. The MIME types have been extended on top of the base *nginx* MIME types with WASM support.
```
```suggestion
Below contains a performance tuned version of *nginx* fitting WebAssembly deployments. The MIME types have been extended on top of the base *nginx* MIME types with WASM support.
```

jeromelaban reviewed (commented):
Thanks for your contribution! Could you move the file to https://github.com/unoplatform/uno/tree/master/doc/articles, and name if "how-to-host-a-webassembly-app.md"

RanadeepPolavarapu commented:
> Thanks for your contribution! Could you move the file to https://github.com/unoplatform/uno/tree/master/doc/articles, and name if "how-to-host-a-webassembly-app.md"

@jeromelaban Thank you for the feedback. I have incorporated your suggestions and also moved & renamed the document.

jeromelaban reviewed (approved):
Thank you for your contribution !
